pkg/planner/core/planbuilder.go

```diff
@@ -4239,8 +4241,32 @@ func (b *PlanBuilder) buildImportInto(ctx context.Context, ld *ast.ImportIntoStm
 		return nil, err
 	}
 
-	outputSchema, outputFields := convert2OutputSchemasAndNames(importIntoSchemaNames, importIntoSchemaFTypes)
-	p.setSchemaAndNames(outputSchema, outputFields)
+	if ld.Select != nil {
+		// privilege of tables in select will be checked here
+		selectPlan, err2 := b.Build(ctx, ld.Select)
+		if err2 != nil {
+			return nil, err2
+		}
+		// it's allowed to use IMPORT INTO t FROM SELECT * FROM t
+		// as we pre-check that the target table must be empty.
+		if (len(ld.ColumnsAndUserVars) > 0 && len(selectPlan.Schema().Columns) != len(ld.ColumnsAndUserVars)) ||
+			(len(ld.ColumnsAndUserVars) == 0 && len(selectPlan.Schema().Columns) != len(tableInPlan.VisibleCols())) {
+			return nil, ErrWrongValueCountOnRow.GenWithStackByArgs(1)
+		}
+		for _, cu := range ld.ColumnsAndUserVars {
+			// all fields must be column
```

Comment:
Does IMPORT INTO without SELECT support user variables? If not we can find the error when parsing SQL.

Reply:
https://github.com/pingcap/tidb/pull/49976/commits/5774e74132e3b0226b03194c036939b98594455e

Reply:
maybe use `ColumnNameList` (need to add brackets) instead of `ColumnNameOrUserVarListOptWithBrackets`